app/models/generic_event.rb

```diff
@@ -133,4 +149,14 @@ def self.relationship_attributes(*attributes)
       end
     end
   end
+
+  def self.eventable_types(*types)
```

Comment:
Already quite a big PR. Going to refactor other events to use this method instead of the concern.